src/GithubLink.js

```diff
@@ -0,0 +1,51 @@
+import { PJAX_LINK_ID } from 'constants/github'
+
+const getHrefPathname = (href = '') => {
+  try {
+    const url = new URL(href)
+    return url.pathname
+  } catch {
+    const url = new URL(href, 'https://github.com')
+    return url.pathname
+  }
+}
+
+/**
+ * Component for navigation to Github page without refresh.
+ *
+ * @note These approach is based on `PJAX` attribute, but sometimes failed in some cases
+ *       e.g. user page <=> other pages
+ * @example
+ *   <GithubLink href="/user/repo/">repo index</GithubLink>
+ */
+const GithubLink = ({
+  href = '/',
+  disableSPA = false,
+  skipIfPathnameRepeated = true,
+  onClick,
+  children,
+  ...rest
+}) => {
+  const handleClick = (e) => {
+    if (skipIfPathnameRepeated && getHrefPathname(href) === location.pathname) {
```

Comment:
`skipIfPathnameRepeated` aims to prevent navigation after clicking the link on the same page, not sure if there is any better naming for this variable.